client.go

```diff
@@ -52,190 +52,184 @@ func newClient(origin, ignoreDestCSV string) (*client, error) {
 	}, nil
 }
 
-func (c *client) watchNamespaceEvents(ctx context.Context, wg *sync.WaitGroup) error {
+func (c *client) watchNamespaceEvents(ctx context.Context, wg *sync.WaitGroup) (stop bool, err error) {
+	wg.Add(1)
+	defer wg.Done()
+
 	watcher, err := c.clientset.CoreV1().Namespaces().Watch(metav1.ListOptions{})
 	if err != nil {
-		return fmt.Errorf("failed to create watch interface: %s", err)
+		return true, fmt.Errorf("failed to create watch interface: %s", err)
 	}
 
 	// ADDED events are also notified for namespaces that already exist at startup.
 	// Therefore, necessary to ignore this notification only at startup.
 	namespaces, err := c.clientset.Namespaces().List(metav1.ListOptions{})
 	if err != nil {
-		return fmt.Errorf("failed to receive a namespace list: %s", err)
+		return true, fmt.Errorf("failed to receive a namespace list: %s", err)
 	}
 	existingNS := map[string]int{}
 	for _, ns := range namespaces.Items {
 		existingNS[ns.ObjectMeta.Name] = 0 // meaningless value
 	}
 
 	// Event loop
-	go func() {
-		wg.Add(1)
-		for {
-			select {
-			case <-ctx.Done():
-				watcher.Stop()
-				wg.Done()
-				return
-			case ev := <-watcher.ResultChan():
-				if ev.Object == nil {
-					// Closed because of error
-					return
-				}
+	for {
+		select {
+		case <-ctx.Done():
+			watcher.Stop()
+			return true, nil
+		case ev := <-watcher.ResultChan():
+			if ev.Object == nil {
+				// Closed because of timeout
+				return false, nil
+			}
 
-				if ev.Type != watch.Added {
-					continue
-				}
+			if ev.Type != watch.Added {
+				continue
+			}
+
+			ns := ev.Object.(*apiv1.Namespace)
 
-				ns := ev.Object.(*apiv1.Namespace)
+			newNS := ns.ObjectMeta.Name
 
-				newNS := ns.ObjectMeta.Name
+			// Skip existing namaspaces only at startup
+			if _, ok := existingNS[newNS]; ok {
+				delete(existingNS, newNS)
+				continue
+			}
+
+			secrets, err := c.clientset.Secrets(c.origin).List(metav1.ListOptions{})
+			if err != nil {
+				log.Printf("failed to receive a secret list from '"+c.origin+"': %s\n", err)
+				continue
+			}
 
-				// Skip existing namaspaces only at startup
-				if _, ok := existingNS[newNS]; ok {
-					delete(existingNS, newNS)
+			// Copy secret from K8NSKEL_ORIGIN
+			for _, secret := range secrets.Items {
+				// Skip the default token
+				if strings.HasPrefix(secret.Name, defaultTokenPrefix) && secret.Type == apiv1.SecretTypeServiceAccountToken {
 					continue
 				}
 
-				secrets, err := c.clientset.Secrets(c.origin).List(metav1.ListOptions{})
+				_, err := c.clientset.Secrets(newNS).Create(&apiv1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secret.ObjectMeta.Name,
+						Namespace: newNS,
+					},
+					Type: secret.Type,
+					Data: secret.Data,
+				})
 				if err != nil {
-					log.Printf("failed to receive a secret list from '"+c.origin+"': %s\n", err)
+					log.Printf("failed to copy a secret '"+secret.ObjectMeta.Name+"' from '"+c.origin+"' to "+newNS+": %s\n", err)
```

Comment:
Please use `%s` instead of String concatenation.

Reply:
Fixed!